task/sonarqube-scanner/0.1/sonarqube-scanner.yaml

```diff
@@ -28,8 +28,20 @@ spec:
     - name: SONAR_PROJECT_KEY
       description: Project's unique key
       default: ""
+    - name: SONAR_LOGIN_SECRET_NAME
+      description: Sonar login secret name
+      default: ""
+    - name: SONAR_LOGIN_SECRET_KEY
+      description: Sonar login secret key
+      default: ""
   steps:
     - name: sonar-properties-create
+      env:
+        - name: SONAR_LOGIN
+          valueFrom:
+            secretKeyRef:
+              name: $(params.SONAR_LOGIN_SECRET_NAME)
```

Comment:
I think this change might fail if the user doesn't provides the following param values:
- `SONAR_LOGIN_SECRET_NAME`
- `SONAR_LOGIN_SECRET_KEY` 

which defaults to `" "` and which will give following error when we run the task:

```
[spec.containers[0].env[1].valueFrom.secretKeyRef.name: Invalid value:
"": a DNS-1123 subdomain must consist of lower case alphanumeric characters,
''-'' or ''.'', and must start and end with an alphanumeric character
(e.g. ''example.com'', regex used for validation is ''[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*''),
```

Another workaround for this problem can be to use [optional workspace](https://meet.google.com/linkredirect?authuser=0&dest=https%3A%2F%2Fgithub.com%2Ftektoncd%2Fpipeline%2Fblob%2Fmaster%2Fdocs%2Fworkspaces.md%23optional-workspaces) for mounting the secrets and avoid setting the `env`.

Reply:
How can I mount secret to the workspace?

Reply:
Probably in task you can have workspace like this:
```
workspaces:
    - name: secrets
```
and in the script of the task you can just try:

```
cat ${workspaces.secrets.path}/token
```
and in the taskrun you can mount secrets like:
```
workspaces:
    - name: secrets
      secret:
        secretName: login-credentials
```

Reply:
In this case what if the login-credential is `""` as above? Will it fail with the same error message or it has different behavior?

And I didn't see the secret in spec.

<img width="1024" alt="Screenshot 2020-11-25 at 11 29 29 AM" src="https://user-images.githubusercontent.com/29942725/100195927-7d6a4c80-2f11-11eb-815c-e66d2eeebbdd.png">

Reply:
If you don't want to mount secrets then `optional workspaces` would be helpful in that case :)

Reply:
Got it thanks

Reply:
I fixed let me test on my cluster and get back to you